setup/install_gatling/30_classification/50_classify_nodes_via_site_pp.rb

```diff
@@ -0,0 +1,42 @@
+test_name "Classify Puppet agents on master"
+skip_test 'Installing PE, not FOSS' unless ENV['BEAKER_INSTALL_TYPE'] == 'foss'
+
+def generate_sitepp()
+  "node /.*agent.*/ { include #{ENV['PUPPET_SCALE_CLASS']} }\nnode /#{agent.hostname}/ { include #{ENV['PUPPET_SCALE_CLASS']} }\nnode 'default' {}"
+end
+
+def classify_foss_nodes(host)
+  environments = on(host, puppet('config print environmentpath')).stdout.chomp
+  # on old PEs (3.3 at least), directory environments have to be enabled. We
+  # can detect this situation because the command above will return an empty
+  # string.  In that case we'll take a swing at enabling them by modifying the
+  # puppet config file, and then running the command again.
+  #
+  # This may need to be changed to support more versions of PE and/or OSS in
+  # the future.
+  if environments == ""
+    on(host, puppet('config set environmentpath \$confdir/environments'))
+    # need to restart the server to make this change take effect.
+    # TODO: should move the restart into a helper method or something, this
+    # is duplicated in 99_restart_server.rb.
+    service_name = get_puppet_server_service_name_from_env()
+
+    on(host, "systemctl restart #{service_name}")
```

Comment:
i think there is a beaker builtin for this (i think it gets called by `#with_puppet_running_on`)?  and i think unless you tell it otherwise it will HUP instead of restart.  much faster.